src/status_im2/contexts/chat/composer/effects.cljs

```diff
@@ -106,21 +106,34 @@
 (defn use-edit
   [{:keys [input-ref]}
    {:keys [text-value saved-cursor-position]}
-   {:keys [edit]}]
+   {:keys [edit input-with-mentions]}]
   (rn/use-effect
    (fn []
-     (let [edit-text        (get-in edit [:content :text])
-           text-value-count (count @text-value)]
+     (let [mention?         (some #(= :mention (first %)) (seq input-with-mentions))
+           mention-text     (reduce (fn [acc item]
+                                      (str acc (second item)))
+                                    ""
+                                    input-with-mentions)
+           edit-text        (if mention? mention-text (get-in edit [:content :text]))
+           text-value-count (count @text-value)
+           inject-edit-text (fn []
+                              (reset! text-value edit-text)
+                              (reset! saved-cursor-position (if (zero? text-value-count)
+                                                              (count edit-text)
+                                                              text-value-count))
+                              (.setNativeProps ^js @input-ref (clj->js {:text edit-text})))]
```

Comment:
Just mentioning since you're touching this region of text, not a suggestion for change. Calling methods on atoms that are initialized as `nil` can be dangerous, we've seen people had problems with this because the ref will only be set after one render pass, so we usually need to to wrap them in `(when @input-ref ...)`.